nimble/host/include/host/ble_gatt.h

```diff
@@ -320,7 +320,7 @@ int ble_gattc_read_by_uuid(uint16_t conn_handle, uint16_t start_handle,
  * @param cb_arg                The optional argument to pass to the callback
  *                                  function.
  *
- * @return                      0 on success; nonzero on failure.
+ * @return                      BLE_HS_EDONE on success; nonzero on failure.
```

Comment:
why do you need to change success return value?

Reply:
because there is no partial reading , the full reading is returned when returning BLE_HS_EDONE

Reply:
IMO...

When possible, functions should return 0 on success.  `BLE_HS_EDONE` is a "necessary evil" for when there are multiple ways to succeed.  In this case, success can mean only one thing, so we should use 0.

Reply:
@ccollins476ad updated.